File: medianof2Arrays.java
class Solution {
    public double findMedianSortedArrays(int[] nums1, int[] nums2) {
        int l1=0;
        int l2=0;
        int m=nums1.length;
        int n=nums2.length;
        double median=0;
        int[] temp=new int[m+n];
        int k=0;
     while ( l1< m && l2 < n) {
      if (nums1[l1] <= nums2[l2]) {
        temp[k++] = nums1[l1++];
      } else {
        temp[k++] = nums2[l2++];
       
      }
    }

    // When we run out of elements in either L or M,
    // pick up the remaining elements and put in A[p..r]
    while (l1 < m) {
      temp[k++] = nums1[l1++];
      
    }

    while (l2 < n) {
      temp[k++] = nums2[l2++];
    }
  
        for(int j:temp){
            System.out.print(j);
        }
        if((m+n)%2==0){
            // length is 4..get 1 and 2 index 
            median=(double)(temp[(m+n)/2-1]+temp[(m+n)/2])/2;
        }
       else 
           median=(double)temp[(m+n)/2];
 
     return median;      
}
}

File: MedianOf2Arrays.py
#All TC on leetcode passed


class Solution:
    def findMedianSortedArrays(self, nums1: List[int], nums2: List[int]) -> float:

        #Here we use binary search to divide the array into halves.
        #Time complexity - log(m) where m is length of shorter array
        #Space compexity - O(1) 

        
        #swapping to perform B.S. on shorter array
        if len(nums1)>len(nums2):
            nums1, nums2 = nums2, nums1

        m = len(nums1)
        n = len(nums2)

        l = 0
        h = m-1

        while True:

            mid1 = (l+h)//2
            mid2 = ((m+n)//2) - (mid1+1) -1 #gives mid2 idx in nums2

            l1 = nums1[mid1] if mid1>=0 else -math.inf
            r1 = nums1[mid1+1] if (mid1+1)<m else math.inf
            l2 = nums2[mid2] if mid2>=0 else -math.inf
            r2 = nums2[mid2+1] if (mid2+1)<n else math.inf

            if l1<=r2 and l2<=r1:
                if ((m+n)%2==0):
                    return (max(l1, l2) + min(r1, r2))/2
                else:
                    return min(r1, r2)        
            elif l1>r2:
                h = mid1-1
            else:
                l = mid1+1

        return -1


       
      
       